consensus/hotstuff/votecollector/base_vote_collector.go

```diff
@@ -0,0 +1,22 @@
+package votecollector
+
+import (
+	"github.com/gammazero/workerpool"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type BaseVoteCollector struct {
```

Comment:
I get your point why we need the BaseVoteCollection, as it contains the shared data/resource for vote collectors in different status.
And because VoteCollector could be in either caching or verifying or invalid status, regardless which status
the vote collector is in, they all need the blockID and a pool of workers to be able process votes
concurrently. 

However, the workers for different status work differently. I'm not sure whether we want to share the pool. 

In my opinion, we should let vote collector in different status own their own worker pool. When switching from caching to verifying status, we probably don't want to share the workers, but actually stop all workers for caching status ?

But maybe I'm wrong, if we stop all workers from caching status, then we are equivalently dropping the votes that are currently processed by those workers, right? 

If we don't want to stop the workers, then we will need to allow the worker to process votes when the vote collector is changing status. Is that the reason a shared worker pool would be easier to do?

Reply:
I think that pool will be used only by verifying collector. This base collector can contain all shared data, same as [here](https://github.com/onflow/flow-go/blob/5e6ef3da859f422c762ddeafc597dc67044a58a5/engine/consensus/approvals/assignment_collector_base.go#L20). 
I think it's really ok to have a shared pool of workers for all vote processing, remember that we don't want to spawn too many goroutines.